x/stakeibc/keeper/msg_server_liquid_stake.go

```diff
@@ -17,114 +15,91 @@ import (
 func (k msgServer) LiquidStake(goCtx context.Context, msg *types.MsgLiquidStake) (*types.MsgLiquidStakeResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// Init variables
-	// deposit `amount` of `denom` token to the stakeibc module
-	// NOTE: Should we add an additional check here? This is a pretty important line of code
-	// NOTE: If sender doesn't have enough inCoin, this errors (error is hard to interpret)
-	// check that hostZone is registered
-	// strided tx stakeibc liquid-stake 100 uatom
+	// Get the host zone from the base denom in the message (e.g. uatom)
 	hostZone, err := k.GetHostZoneFromHostDenom(ctx, msg.HostDenom)
 	if err != nil {
-		k.Logger(ctx).Error(fmt.Sprintf("Host Zone not found for denom (%s)", msg.HostDenom))
-		return nil, errorsmod.Wrapf(types.ErrInvalidHostZone, "no host zone found for denom (%s)", msg.HostDenom)
-	}
-	// get the sender address
-	sender, _ := sdk.AccAddressFromBech32(msg.Creator)
-	// get the coins to send, they need to be in the format {amount}{denom}
-	// is safe. The converse is not true.
-	ibcDenom := hostZone.GetIbcDenom()
-	coinString := msg.Amount.String() + ibcDenom
-	inCoin, err := sdk.ParseCoinNormalized(coinString)
-	if err != nil {
-		k.Logger(ctx).Error(fmt.Sprintf("failed to parse coin (%s)", coinString))
-		return nil, errorsmod.Wrapf(err, "failed to parse coin (%s)", coinString)
+		return nil, errorsmod.Wrapf(types.ErrInvalidToken, "no host zone found for denom (%s)", msg.HostDenom)
 	}
 
-	// Creator owns at least "amount" of inCoin
-	balance := k.bankKeeper.GetBalance(ctx, sender, ibcDenom)
-	if balance.IsLT(inCoin) {
-		k.Logger(ctx).Error(fmt.Sprintf("balance is lower than staking amount. staking amount: %v, balance: %v", msg.Amount, balance.Amount))
-		return nil, errorsmod.Wrapf(sdkerrors.ErrInsufficientFunds, "balance is lower than staking amount. staking amount: %v, balance: %v", msg.Amount, balance.Amount)
+	// Get user and module account addresses
+	userAddress, err := sdk.AccAddressFromBech32(msg.Creator)
```

Comment:
nit: sender is clearer to me, but I don't have a strong view! other addresses (e.g. modules) could stake, whereas `userAddress` sounds like an individual

Reply:
Agreed `user` isn't great, but I think sender is a bit too generic considering there are multiple sends going on in this function. Can you think of a good third option?

Reply:
`msgSender` perhaps?

Reply:
`caller`, `liquidStakingAddress` could also work

Reply:
I like those last two! How bout `liquidStakerAddress`?

Reply:
sgtm!

Reply:
done